### PR TITLE
Multiple changes to deploy scripts

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

1. Update Origin to v 3.10
2. Make local config dir configurable
3. Make it possible to pass OKD components, like ansible-service-broker. Important for Che APB testing
4. Make it possible to add cluster-admin role to developer user (currently required to run Che APB)
5. Edit ansible-service-broker config map (admin sandbox role required)
6. Make it possible to override ansible-service-broker image and tags (currently I have issues with latest)
7. Make it possible to override Keycloak image, tag and pull policy (helpful when building custom Keycloak image to be tested with Che multiuser)

### How to try Che APB?

```
./ocp.sh --run-ocp --enable-apb --admin
```
Che APB will be available in the catalog:

image

### CODE REVIEW

**Comment by eivantsov** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/584/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/584//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by eivantsov** [comment]
ci-test

File: deploy/openshift/ocp.sh [hunk 1]
```diff
@@ -162,10 +172,58 @@ wait_ocp() {
...
+            available=$(${OC_BINARY} get dc/openshift-automation-service-broker -n=openshift-automation-service-broker -o json | jq ".status.conditions[] | select(.type == \"Available\") | .status")
+            progressing=$(${OC_BINARY} get dc/openshift-automation-service-broker -n=openshift-automation-service-broker -o json | jq ".status.conditions[] | select(.type == \"Progressing\") | .status")
+            timeout_in=$((end-SECONDS))
+            echo "Deployment is in progress...(Available.status=${available}, Progressing.status=${progressing}, ${timeout_in} seconds remain)"
+            sleep ${POLLING_INTERVAL_SEC}
+        done
+        if [ "${progressing}" == "\"True\"" ]; then
+            echo "Ansible service catalog deployed successfully"
+        elif [ "${progressing}" == "False" ]; then
+            echo "Ansible service catalog deployment failed. Aborting."
```
Comment (sleshchenko) [inline deploy/openshift/ocp.sh:206]: Should exit 1 be called for Aborting?

Comment (ghost) [reply deploy/openshift/ocp.sh:206]: Fixed

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/585/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/585//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by eivantsov** [comment]
ci-build

**Comment by riuvshin** [comment]
ci-test

File: deploy/openshift/ocp.sh [hunk 2]
```diff
@@ -162,10 +172,59 @@ wait_ocp() {
...
+    if [ "${APB}" == "true" ]; then
+        export  ENABLE_COMPONENTS="--enable=service-catalog,router,registry,web-console,persistent-volumes,rhel-imagestreams,automation-service-broker"
+        echo "[OCP] start OKD with '${ENABLE_COMPONENTS}'"
+    fi
+    $OC_BINARY cluster up --public-hostname="${OC_PUBLIC_HOSTNAME}" \
+                          --routing-suffix="${OC_PUBLIC_IP}.${DNS_PROVIDER}" \
+                          --base-dir=${OKD_DIR} \
+                          "${ENABLE_COMPONENTS}"
     wait_ocp
+#    wait_for_automation_service_broker
```
Comment (dmytro-ndp) [inline deploy/openshift/ocp.sh]: Didn't you forget to uncomment this line?

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/594/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/594//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by dmytro-ndp** [comment]
I see an error in selenium tests report:
```
error: dial tcp 172.30.3.247:8443: getsockopt: connection refused - verify you have provided the correct host and port and that the server is currently running.
Test nip.io - works OK, using it as DNS provider
```
@eivantsov: does it make sense to run E2E tests again after  `wait_for_automation_service_broker` had been uncommited?

**Comment by eivantsov** [comment]
@dmytro-ndp the problem is that verison 3.10 does not run on CI.

My changes are about adding new functionality to script that tests do not use, and changing version.

@riuvshin and myself will take a look

Of course, it won't be merged until tests pass.

**Comment by dmytro-ndp** [comment]
I see. Thank you for explanation.

**Comment by riuvshin** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/599/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/599//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by riuvshin** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/607/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/607//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by riuvshin** [comment]
ci-test

**Review by riuvshin — APPROVED** [review]

**Review by riuvshin — APPROVED** [review]
please do not merge this before https://gitlab.cee.redhat.com/che/infra/issues/196 is done

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/609/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/609//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by dmytro-ndp** [comment]
There are strange error of starting workspace http://tests.codenvycorp.com/selenium_test_report_data/10766/che_server_logs/che-server-pod.log
```
2018-08-16 13:34:27,632[nio-8080-exec-8]  [INFO ] [o.e.c.a.w.s.WorkspaceManager 469]    - Workspace 'user1534424686218/workspaceiobif4' with id 'workspaceq3wluvwl9hulvihw' created by user 'user1534424686218'
2018-08-16 13:34:28,271[io-8080-exec-10]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 330]   - Starting workspace 'user1534424686218/workspaceiobif4' with id 'workspaceq3wluvwl9hulvihw' by user 'user1534424686218'
2018-08-16 13:34:28,317[aceSharedPool-1]  [WARN ] [.i.k.KubernetesInternalRuntime 193]  - Failed to start Kubernetes runtime of workspace workspaceq3wluvwl9hulvihw. Cause: Failure executing: POST at: https://172.29.4.88:8443/apis/route.openshift.io/v1/namespaces/eclipse-che/routes. Message: 404 page not found
.
2018-08-16 13:34:28,370[aceSharedPool-1]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 391]   - Workspace 'user1534424686218:workspaceiobif4' with id 'workspaceq3wluvwl9hulvihw' start failed
```
-------
```
2018-08-16 13:34:06,213[nio-8080-exec-3]  [INFO ] [o.e.c.a.w.s.WorkspaceManager 469]    - Workspace 'user1534424686218/workspacelk9x6t' with id 'workspacet7ch7h7sxyk6hw2a' created by user 'user1534424686218'
2018-08-16 13:34:06,832[nio-8080-exec-3]  [ERROR] [o.e.c.a.w.s.WorkspaceRuntimes 345]   - Failure executing: POST at: https://172.29.4.88:8443/apis/project.openshift.io/v1/projectrequests. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. projectrequests.project.openshift.io is forbidden: User "system:serviceaccount:eclipse-che:che" cannot create projectrequests.project.openshift.io at the cluster scope: You may not request a new project via this API.. The error may be caused by an expired token or changed password. Update Che server deployment with a new token or password.
org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInfrastructureException: Failure executing: POST at: https://172.29.4.88:8443/apis/project.openshift.io/v1/projectrequests. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. projectrequests.project.openshift.io is forbidden: User "system:serviceaccount:eclipse-che:che" cannot create projectrequests.project.openshift.io at the cluster scope: You may not request a new project via this API.. The error may be caused by an expired token or changed password. Update Che server deployment with a new token or password.
	at org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProject.create(OpenShiftProject.java:119)
	at org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProject.prepare(OpenShiftProject.java:88)
	at org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProjectFactory.create(OpenShiftProjectFactory.java:58)
	at org.eclipse.che.workspace.infrastructure.openshift.OpenShiftRuntimeContext.getRuntime(OpenShiftRuntimeContext.java:65)
	at org.eclipse.che.workspace.infrastructure.openshift.OpenShiftRuntimeContext.getRuntime(OpenShiftRuntimeContext.java:29)
	at org.eclipse.che.api.workspace.server.WorkspaceRuntimes.startAsync(WorkspaceRuntimes.java:317)
	at org.eclipse.che.api.workspace.server.WorkspaceManager.startAsync(WorkspaceManager.java:374)
	at org.eclipse.che.api.workspace.server.WorkspaceManager.startWorkspace(WorkspaceManager.java:291)
	at org.eclipse.che.multiuser.resource.api.workspace.LimitsCheckingWorkspaceManager.startWorkspace(LimitsCheckingWorkspaceManager.java:119)
	at org.eclipse.che.api.workspace.server.WorkspaceService.startById(WorkspaceService.java:340)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:140)
	at org.everrest.core.impl.method.DefaultMethodInvoker.invokeMethod(DefaultMethodInvoker.java:60)
	at org.everrest.core.impl.RequestDispatcher.doInvokeResource(RequestDispatcher.java:306)
	at org.everrest.core.impl.RequestDispatcher.invokeSubResourceMethod(RequestDispatcher.java:297)
	at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:233)
	at org.everrest.core.impl.RequestDispatcher.dispatch(RequestDispatcher.java:128)
	at org.everrest.core.impl.RequestHandlerImpl.handleRequest(RequestHandlerImpl.java:62)
	at org.everrest.core.impl.EverrestProcessor.process(EverrestProcessor.java:120)
	at org.everrest.core.servlet.EverrestServlet.service(EverrestServlet.java:61)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
	at com.google.inject.servlet.ServletDefinition.doServiceImpl(ServletDefinition.java:286)
	at com.google.inject.servlet.ServletDefinition.doService(ServletDefinition.java:276)
	at com.google.inject.servlet.ServletDefinition.service(ServletDefinition.java:181)
	at com.google.inject.servlet.ManagedServletPipeline.service(ManagedServletPipeline.java:91)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:85)
	at org.eclipse.che.commons.logback.filter.IdentityIdLoggerFilter.doFilter(IdentityIdLoggerFilter.java:53)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.multiuser.keycloak.server.KeycloakEnvironmentInitalizationFilter.doFilter(KeycloakEnvironmentInitalizationFilter.java:105)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.multiuser.keycloak.server.KeycloakAuthenticationFilter.doFilter(KeycloakAuthenticationFilter.java:121)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.multiuser.machine.authentication.server.MachineLoginFilter.doFilter(MachineLoginFilter.java:129)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.eclipse.che.commons.logback.filter.RequestIdLoggerFilter.doFilter(RequestIdLoggerFilter.java:50)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at org.apache.catalina.filters.CorsFilter.handleNonCORS(CorsFilter.java:423)
	at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:169)
	at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:82)
	at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:120)
	at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:135)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at ch.qos.logback.access.tomcat.LogbackValve.invoke(LogbackValve.java:256)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:677)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:803)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:868)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1459)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://172.29.4.88:8443/apis/project.openshift.io/v1/projectrequests. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. projectrequests.project.openshift.io is forbidden: User "system:serviceaccount:eclipse-che:che" cannot create projectrequests.project.openshift.io at the cluster scope: You may not request a new project via this API..
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:472)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:409)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:381)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:227)
	at io.fabric8.openshift.client.dsl.internal.ProjectRequestsOperationImpl.create(ProjectRequestsOperationImpl.java:74)
	at io.fabric8.openshift.client.dsl.internal.ProjectRequestsOperationImpl$1.apply(ProjectRequestsOperationImpl.java:91)
	at io.fabric8.openshift.client.dsl.internal.ProjectRequestsOperationImpl$1.apply(ProjectRequestsOperationImpl.java:87)
	at io.fabric8.openshift.api.model.DoneableProjectRequest.done(DoneableProjectRequest.java:27)
	at org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProject.create(OpenShiftProject.java:116)
	... 62 common frames omitted
2018-08-16 13:34:06,892[er-ThreadPool-1]  [ERROR] [e.c.w.i.k.n.p.PVCSubPathHelper 155]  - Unable to perform '[rm, -rf, /tmp/job_mount/workspacet7ch7h7sxyk6hw2a]' command for the workspace 'workspacet7ch7h7sxyk6hw2a' cause: 'Failure executing: POST at: https://172.29.4.88:8443/apis/project.openshift.io/v1/projectrequests. Message: Forbidden!Configured service account doesn't have access. Service account may have been revoked. projectrequests.project.openshift.io is forbidden: User "system:serviceaccount:eclipse-che:che" cannot create projectrequests.project.openshift.io at the cluster scope: You may not request a new project via this API.. The error may be caused by an expired token or changed password. Update Che server deployment with a new token or password.'
```

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/615/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/615//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/628/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/628//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by riuvshin** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/632/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/632//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/10766)
docker image: eclipseche/che-server:10766
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by SkorikSergey** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/615/) doesn't show any regression against this Pull Request.

**Comment by dmytro-ndp** [comment]
I had encountered an error on OCP with OCP oauth ran with command  `/home/${USER}/projects/che/deploy/openshift/ocp.sh --run-ocp --deploy-che --multiuser --setup-ocp-oauth`:
```
che-selenium-test/selenium-tests.sh --test=org.eclipse.che.selenium.site.ocpoauth.** --multiuser --host=che-eclipse-che.172.19.20.137.nip.io --port=80
...
[INFO] Running TestSuite
2018-08-21 18:56:39,527[main]             [INFO ] [UserCheDefaultTestUserProvider 69]   - User name='admin', id='b07e3a58-ed50-4a6e-be17-fcf49ff8b242' is being used as admin
2018-08-21 18:56:44,081[main]             [INFO ] [.c.s.c.c.k.c.KeycloakCliClient 80]   - Default test user with name='user1534866998179' and id='b8a1d0ec-b90b-4910-ada1-2139def13813' has been created.
2018-08-21 18:56:48,502[main]             [INFO ] [.c.s.c.c.k.c.KeycloakCliClient 80]   - Default test user with name='user1534867004093' and id='6ecb5553-6eaf-414a-81e7-b50ccd719c74' has been created.
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-failsafe-plugin:2.20.1:verify (default) @ che-selenium-test ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 21.524 s
[INFO] Finished at: 2018-08-21T18:56:48+03:00
[INFO] Final Memory: 55M/1217M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.20.1:verify (default) on project che-selenium-test: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/ndp/projects/che/selenium/che-selenium-test/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] Unable to create injector, see the following errors:
[ERROR] 
[ERROR] 1) Error injecting constructor, java.lang.RuntimeException: Failed to get access token for user 'user1534866998179'
```

**Comment by dmytro-ndp** [comment]
There was wrong value of _CHE_INFRASTRUCTURE_ variable was set. Sorry - my fault.
With correct value tests from the package **org.eclipse.che.selenium.site.ocpoauth** have been passed successfully.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**